### DISCUSSION TITLE
CODER ubuntu 20x Error SSL

### DISCUSSION THREAD

**Discussion by nomopo** [coder/code-server#4648] [Q&A]
Hello, i do all and install in ubuntu server 20.x and it works fine with http:// my problem is when i install ssl, it doesn't work, i can run this. i have a page not found. 

I change port to 8888 in http and work's fine. But when i create ssl, that it's created succfully, and redirect all to https, don't work anythink.

Please can yo u help me? thanks a lot

**Comment by nomopo** [comment]
If i run http://sub.domain.com:8888 it work's fine but if i like to put https, don't works,

**Reply by jsjoeio** [reply]
I wonder if it has to do with the port in the name on HTTPS? But I have no idea. @code-asher any ideas?

**Reply by nomopo** [reply]
> I wonder if it has to do with the port in the name on HTTPS? But I have no idea. @code-asher any ideas?

I reinstall ssl and now it works with https://sub.domain.com:8888 but without ssl security.

**Reply by jsjoeio** [reply]
Hmmm...I don't have enough experience to help much with SSL issues 🤔 So you're saying code-server works as expected, but doesn't have SSL security?

**Reply by nomopo** [reply]
Yes, now i can't use it with ssl, i install certificate but it doesn't work in port 8888.

**Comment by code-asher** [comment]
How are you setting up the certificate?  Is it through code-server's
--cert flag or through a reverse proxy?

**Comment by nomopo** [comment]
I install in server certbot --nginx -d sub.domain.com -m mail@mail.com

but when i open https://sub.domain.com:443 and it doesn't work, or ir show me https://xxx is not secure. But if i write certbot, i can see sub.domain.com with agree ssl.

**Reply by code-asher** [reply]
Hmm maybe NGINX needs to be reloaded?  Although I think certbot is supposed to reload it for you.  What does your NGINX config look like?

Maybe there will be something helpful in our docs: https://github.com/coder/code-server/blob/main/docs/guide.md#using-lets-encrypt-with-nginx

**Comment by nomopo** [comment]
I format all and begin from 0. And when i install ssl i have this result:

502 Bad Gateway
nginx/1.18.0 (Ubuntu)





El 20 de diciembre de 2021 a las 17:55:12, Asher ***@***.******@***.***>) escribió:

Hmm maybe NGINX needs to be reloaded? Although I think certbot is supposed to reload it for you. What does your NGINX config look like?

Maybe there will be something helpful in our docs: https://github.com/coder/code-server/blob/main/docs/guide.md#using-lets-encrypt-with-nginx

—
Reply to this email directly, view it on GitHub<https://github.com/coder/code-server/discussions/4648#discussioncomment-1848319>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAM47WI62LRNGK3KONN4YTDUR5NW3ANCNFSM5KJNN4WA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by nomopo** [comment]
```
server {
    server_name sub.domain.net;

    location / {
      proxy_pass http://localhost:8080/;
      proxy_set_header Host $host;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection upgrade;
      proxy_set_header Accept-Encoding gzip;
    }

    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/code.nredes.net/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/code.nredes.net/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot


}
server {
    if ($host = sub.domain.net) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    listen 80;
    listen [::]:80;
    server_name sub.domain.net;
    return 404; # managed by Certbot


}
```

**Reply by code-asher** [reply]
Bugger, this looks completely fine as far as I can tell.  You are going to `https://sub.domain.net` in the browser and getting a 502? 

You said you already confirmed it works with `http://sub.domain.net` right?  Something like:

```
server {
    listen [::]:80;
    listen 80;
    server_name sub.domain.net;

    location / {
      proxy_pass http://localhost:8080/;
      proxy_set_header Host $host;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection upgrade;
      proxy_set_header Accept-Encoding gzip;
    }
}
```

I think the next step will be to investigate the NGINX error logs to see if it gives more information on the 502 error.

**Reply by code-asher** [reply]
You can also get a 502 when code-server is not running, can you confirm it is running on port 8080?

**Comment by nomopo** [comment]
In port 8080 works fine

--
Noé Moreno

El día 21 de diciembre de 2021 a las 18:13:28, Asher ***@***.******@***.***>) escrito:

You can also get a 502 when code-server is not running, can you confirm it is running on port 8080?

—
Reply to this email directly, view it on GitHub<https://github.com/coder/code-server/discussions/4648#discussioncomment-1854032>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAM47WNV6T2EEKF4H6ZGW2DUSCYTRANCNFSM5KJNN4WA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
